examples/__tests__/test-programmatic-update.ava.js

```diff
@@ -0,0 +1,48 @@
+import { Worker } from "near-workspaces";
+import test from "ava";
+import * as fs from 'fs';
+import * as path from "path";
+
+test.beforeEach(async (t) => {
+  const worker = await Worker.init();
+  const root = worker.rootAccount;
+
+  const ali = await root.createSubAccount("ali");
+
+  const programmaticUpdate = await root.devDeploy("build/programmatic-update.wasm");
+
+  await programmaticUpdate.call(programmaticUpdate, "init", { manager: ali.accountId });
+
+  t.context.worker = worker;
+  t.context.accounts = { root, programmaticUpdate, ali };
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.worker.tearDown().catch((error) => {
+    console.log("Failed to tear down the worker:", error);
+  });
+});
+
+
+test("the contract can update itself", async (t) => {
+  const { ali, programmaticUpdate } = t.context.accounts;
+
+  const code = fs.readFileSync(path.resolve("./build/programmatic-update.wasm"));
+
+  // Ali is the manager, asks the contract to update itself
+
+  // QUESTION: IN WHICH FORMAT SHOULD WE PASS THE WASM CODE AS ARGS? THE CODE IS OF TYPE BUFFER
+  const result = await ali.callRaw(programmaticUpdate, "updateContract", {
+    code
```

Comment:
I think it cannot be in json. As contract is raw bytes, it can contains ascii `"` and ending json string unexpectedly. However, our high level API only takes json serialized contract arguments and this is not customizable.

So, currently it's only possible to use low level APIs to implement your awesome example. Check `test_promise_batch_deploy_call` in tests/src/test_promise_batch_api.js, use `promiseBatchActionDeployContract`, parsing the raw bytes of the as input (passing Uint8Array to callRaw, and that'll be available in contract from `near.input()`)

Reply:
The other way is expanding our high level API, more specifically, in near-bindgen.ts, `_getArgs` use hard coded way to deserialize the arg with `JSON.parse`. Making that customizable with option can handle your case specially!

Reply:
Thanks. Let me try the above low level API solution based on your reference